src/test/Stack/UploadSpec.hs

```diff
@@ -26,3 +27,8 @@ spec = do
     unless osIsWindows $ do
       status <- getFileStatus fp
       (fileMode status .&. 0o777) `shouldBe` 0o600
+
+  it "finds a HACKAGE_KEY env variable" $ do
+    runRIO () maybeGetHackageKey `shouldReturn` Nothing
+    setEnv "HACKAGE_KEY" "api_key"
```

Comment:
Probably won't bite you, but it could be good to `unsetEnv` after the test, just to avoid any pollution.

Reply:
Something like

```hs

withEnv "HACKAGE_KEY" "api_key"
  $ runRIO () maybeGetHackageKey `shouldReturn` Just (HackageKey "api_key")

withEnv :: String -> String -> IO a -> IO a
withEnv k v f = do
  setEnv k v
  f `finally` unsetEnv k
```